src/client/chat.js

```diff
@@ -22,6 +24,22 @@ module.exports = function (client, options) {
   client._lastChatSignature = null
   client._lastRejectedMessage = null
 
+  // 1.20.3+ serializes chat components in chat packets with NBT. Non-chat packets that send messages (like disconnect) still use JSON chat.
+  // NMP API expects a JSON string, so since the schema is mostly the same we can convert the NBT to JSON with a transform to UUID encoding
+  function handleNbtComponent (nbtDataOrString) {
+    if (mcData.supportFeature('chatPacketsUseNbtComponents') && nbtDataOrString) {
+      // UUIDs are encoded in NBT as a 4x i32 array, so convert to a hex string
+      const simplified = nbt.simplify(nbtDataOrString)
+      return JSON.stringify(simplified, (key, val) => {
+        if (key === 'id' && Array.isArray(val)) return uuid.fromIntArray(val)
+        return val
+      })
+    }
+    // already plaintext JSON or empty
+    return nbtDataOrString
+  }
+  client._handleNbtComponent = handleNbtComponent
```

Comment:
why is this exposed privately and then not used?

if we want to actually expose it, it should be exposed publicly (without _ and in API.md)